warp10/src/main/java/io/warp10/script/functions/GET.java

```diff
@@ -85,4 +100,37 @@ public static int computeAndCheckIndex(NamedWarpScriptFunction func, int index,
 
     return index;
   }
+
+  /**
+   * Recursively get elements of a nested list given a list of indices. At the last recursive call, a single element is returned.
+   *
+   * For example, we will have:
+   * recNestedGet(func, nestedList, [ 0, 1, 2 ])
+   * -> recNestedGet(func, nestedList[0], [ 1, 2 ])
+   * -> recNestedGet(func, (nestedList.get(0)).get(1), [ 2 ])
+   * -> return ((nestedList.get(0)).get(1)).get(2)
+   *
+   * @param func        The function that calls this method
+   * @param nestedList  The list to which to get the elements
+   * @param indexList   The list of indices.
+   * @return
+   * @throws WarpScriptException
+   */
+  static Object recNestedGet(NamedWarpScriptFunction func, List nestedList, List<Number> indexList) throws WarpScriptException {
```

Comment:
Make the function non recursive, also removing the copyIndices allocation since you need only track an index to the original list.